eng/tools/typespec-validation/src/rules/sdk-tspconfig-validation.ts

```diff
@@ -252,6 +252,29 @@ export class TspConfigTsMgmtModularPackageNameMatchPatternSubRule extends TspCon
   }
 }
 
+// ----- TS data plane sub rules -----
+export class TspConfigTsDpPackageDirectorySubRule extends TspconfigEmitterOptionsSubRuleBase {
+  constructor() {
+    super("@azure-tools/typespec-ts", "package-dir", new RegExp(/^(?:[a-z]+-)*rest$/));
```

Comment:
`/^(?:[a-z]+-)*-rest$/` it allows `a--rest`, is this expected? 
In the loop of rules, js specify:
![image](https://github.com/user-attachments/assets/41a546aa-7c6f-4315-9314-aff4a5ffe092)
@MaryGao